include/seqan3/alignment/matrix/detail/trace_iterator.hpp

```diff
@@ -0,0 +1,234 @@
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    using value_type = trace_directions; //!< The value type.
+    using reference = value_type; //!< The reference type.
+    using pointer = void; //!< The pointer type.
+    using difference_type = std::ptrdiff_t; //!< The difference type.
+    using iterator_category = std::forward_iterator_tag; //!< Forward iterator tag.
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr trace_iterator() = default; //!< Defaulted.
+    constexpr trace_iterator(trace_iterator const &) = default; //!< Defaulted.
+    constexpr trace_iterator(trace_iterator &&) = default; //!< Defaulted.
+    constexpr trace_iterator & operator=(trace_iterator const &) = default; //!< Defaulted.
+    constexpr trace_iterator & operator=(trace_iterator &&) = default; //!< Defaulted.
+    ~trace_iterator() = default; //!< Defaulted.
+
+    /*!\brief Constructs from the underlying traceback matrix iterator indicating the start of the traceback path.
+     * \param[in] matrix_iter The underlying matrix iterator.
+     */
+    constexpr trace_iterator(matrix_iter_t const matrix_iter) noexcept : matrix_iter{matrix_iter}
+    {
+        set_trace_direction();
+    }
+
+    /*!\brief Constructs from the underlying traceback matrix iterator indicating the start of the traceback path.
+     * \tparam other_matrix_iter_t The underlying matrix iterator type of `other`; the condition
+     *                             `std::Constructible<matrix_iter_t, other_matrix_iter_t>` must evaluate to `true`.
+     * \param[in] other The underlying matrix iterator.
+     */
+    template <TwoDimensionalMatrixIterator other_matrix_iter_t>
+    //!\cond
+        requires std::Constructible<matrix_iter_t, other_matrix_iter_t>
+    //!\endcond
+    constexpr trace_iterator(trace_iterator<other_matrix_iter_t> const other) noexcept :
+        trace_iterator{other.matrix_iter}
+    {}
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    //!\brief Returns the current traceback direction.
+    reference operator*() const noexcept
+    {
+        return cur_dir;
+    }
+
+    //!\brief Returns the current coordinate in two-dimensional space.
+    constexpr matrix_coordinate coordinate() const noexcept
+    {
+        return matrix_iter.coordinate();
+    }
+    //!\}
+
+    /*!\name Arithmetic operators
+     * \{
+     */
+    //!\brief Advances the iterator by one.
+    constexpr trace_iterator & operator++() noexcept
+    {
+        trace_directions old_dir = *matrix_iter;
+
+        assert(old_dir != trace_directions::none);
+
+        if (cur_dir == trace_directions::up)
+        {
+            matrix_iter -= matrix_offset{row_index_type{1}, column_index_type{0}};
+            // Set new traceback direction if last was up_open.
+            if (static_cast<bool>(old_dir & trace_directions::up_open))
+                set_trace_direction(); // Set the current trace_direction
```

Comment:
```suggestion
                set_current_trace_direction(*matrix_iter);
```

Reply:
I think current does not help either. So I changed it as asked but left the interface name as is.